chemprop/cli/train.py

```diff
@@ -1206,7 +1221,9 @@ def evaluate_and_save_predictions(preds, test_loader, metrics, model_output_dir,
         )
     else:
         df_preds = pd.DataFrame(list(zip(*namess, *preds.T)), columns=columns)
-    df_preds.to_csv(model_output_dir / "test_predictions.csv", index=False)
+
+    if not args.dry_run:
+        df_preds.to_csv(model_output_dir / "test_predictions.csv", index=False)
```

Comment:
```suggestion
```
On line 1122 you return from the function if args.dry_run is True. This is before `evaluate_and_save_predictions` is called.